src/types/base.ts

```diff
@@ -85,17 +91,14 @@ export interface Metric {
     | 'restore';
 }
 
-/**
- * A version of the `Metric` that is used with the attribution build.
- */
-export interface MetricWithAttribution extends Metric {
-  /**
-   * An object containing potentially-helpful debugging information that
-   * can be sent along with the metric value for the current page visit in
-   * order to help identify issues happening to real-users in the field.
-   */
-  attribution: {[key: string]: unknown};
-}
+/** The union of supported metric types. */
+export type Metric =
```

Comment:
I don't think we can do this in v3 because I think it would technically be breaking, but given that we're now defining types of each metric, I wonder if we even need the `Metric` type at all? WDYT?

Reply:
The union is needed in enough places (`bindReporter`, `initMetric`, and `ReportCallback` in this file) that it's convenient to have a name for it, otherwise each of those files would have to define their own and would all have to be updated if/when the next metric is added.

I didn't think of it as a breaking change because
- the only places the old `Metric` was used were internal (again `bindReporter`/`initMetric` and the generic `ReportCallback` type) and all the user-exposed functions return specific metrics (which are unchanged)
- The (new) `Metric` union is assignable to `MetricBase` (the old `Metric`), so even if users have fished out the old `Metric` type for use, in many cases it should be usable unchanged.

But there's no reason we have to repurpose the name (like `MetricWithAttribution`, `Metric` is also documented in the readme). I could switch `MetricBase` back to `Metric` and `Metric` could be...`MetricType`? Blanking on a good name, but open to suggestions :)

Reply:
> I didn't think of it as a breaking change because

Yeah, I don't think what you've done is a breaking change. But I do think it would be considered a breaking chance if we renamed `Metric` to `MetricType` (for example) because presumably some people importing this package are also importing the `Metric` type. So that's why I was saying maybe we could make that change at the next major version release (along with any other type changes we want to make).

Reply:
OK, just to be clear because reusing the `Metric` name has only cause me to be confused in this conversation :) I'm suggesting changing this PR so:
- `Metric` stay `Metric` (not `MetricBase`)
- The new type that's a union of the individual metrics be given some other name (e.g. `MetricType`)

Existing imports of `Metric` would be unchanged and the second one is brand new, so wouldn't affect anything, and since it's all internal, there's no reason for anyone to reach in to import it.

So is there still a problem after making that change and what are you suggesting instead?

Reply:
Sure, that SGTM.